### PR TITLE
Migrate builds from Travis to Github Actions

### LINKED ISSUE #910: Migrate to Github Actions
With the [recent changes announced by Travis.CI ](https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing) regarding open source project builds, we should be moving to alternatives and GitHub Actions seems like a good one for now.

### PR DESCRIPTION
**Pull request by lresende**
Fixes #910

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          python -m pip install --upgrade dist/*.whl
+    - name: Log current Python dependencies version
+      run: |
+        pip freeze
+    - name: Log current Enterprise Gateway version
+      run: |
+        jupyter enterprisegateway --help
+    - name: Lint code
+      run: |
+        flake8 enterprise_gateway
```
Comment (kevin-bates) [inline .github/workflows/build.yml:60]: I think this is redundant since `make dist` performs this lint step.

**Review by kevin-bates — COMMENTED** [review]
This is awesome Luciano - thank you!  I just had the one comment about lint.

**Comment by lresende** [comment]
Here is a green build on my branch:
https://github.com/lresende/enterprise_gateway/actions/runs/409957368

I will only take effect for EG once this is merged into master.

**Review by kevin-bates — APPROVED** [review]
Great PR - thanks!

**Pull request merged**